pkg/repositories/models/partition.go

```diff
@@ -0,0 +1,12 @@
+package models
+
+// Main Use cases:
+// 1. Filter artifacts by partition key/val in a dataset from UI [x]
```

Comment:
does this use-case require a separate index?

Reply:
good question, I don't think so because the primary key will be good enough for us to handle searches on `dataset, key, value`. We will lookup the datasetuuid first and then do the search

Reply:
can you verify that this gorm model translates to primary key index orders in the way you expect?